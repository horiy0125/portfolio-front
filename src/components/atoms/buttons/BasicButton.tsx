import React from 'react';
import styled from 'styled-components';
import { classes } from '@horri1520/hori-base-elements/lib/utils/classes';
import { colors } from '../../../styles/variables';
import Icon from '../icons/Icon';

type Props = {
  children: React.ReactNode;
  onClick: (arg: any) => void;
  icon?: React.ReactNode;
  className?: string;
};

const Root = styled.button`
  display: flex;
  align-items: center;
  color: ${colors.defaultBlue};
  fill: ${colors.defaultBlue};
  opacity: 0.9;
  &:hover {
    opacity: 1;
  }
`;

type LabelProps = {
  hasIcon: boolean;
};
const Label = styled.span<LabelProps>`
  margin-left: ${props => (props.hasIcon ? '8px' : '0')};
`;

const BasicButton: React.VFC<Props> = ({
  children,
  onClick,
  icon,
  className,
}) => {
  const hasIcon = Boolean(icon);

  return (
    <Root className={classes(className)} onClick={arg => onClick(arg)}>
      {icon ? <Icon icon={icon} /> : null}
      <Label hasIcon={hasIcon}>{children}</Label>
    </Root>
  );
};

export default BasicButton;
