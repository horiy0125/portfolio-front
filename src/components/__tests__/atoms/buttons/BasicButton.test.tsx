import React from 'react';
import renderer from 'react-test-renderer';
import BasicButton from '../../../atoms/buttons/BasicButton';

describe('components/atoms/buttons/BasicButton', () => {
  describe('if the component has string children', () => {
    it('render children as it is', () => {
      const component = renderer.create(
        <BasicButton onClick={() => {}}>Button</BasicButton>,
      );
      const tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
