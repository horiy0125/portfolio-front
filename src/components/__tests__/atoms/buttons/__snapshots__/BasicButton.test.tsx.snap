// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/buttons/BasicButton if the component has string children render children as it is 1`] = `
<button
  className="sc-gtsrHT fswUjs"
  onClick={[Function]}
>
  <span
    className="sc-dlnjwi lbpwDs"
  >
    Button
  </span>
</button>
`;
